lib/_stream_readable.js

```diff
@@ -211,7 +211,7 @@ function howMuchToRead(n, state) {
   if (state.objectMode)
     return n === 0 ? 0 : 1;
 
-  if (n === null || isNaN(n)) {
+  if (n === null || n !== n) {
```

Comment:
Oh, good point. This may not just be checking it it is `NaN`, but also non-number values.